internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -109,6 +109,7 @@ func expectedProxyContainers(infra *ir.ProxyInfra,
 		},
 		MaxHeapSizeBytes: maxHeapSizeBytes,
 		XdsServerHost:    ptr.To(fmt.Sprintf("%s.%s.svc.%s", config.EnvoyGatewayServiceName, controllerNamespace, dnsDomain)),
+		LocalClusterName: ptr.To(fmt.Sprintf("%s-%s", config.EnvoyPrefix, utils.GetHashedName(infra.Name, 64))),
```

Comment:
my preference would be to reuse the same naming scheme as service cluster to avoid introducing a new name here

Reply:
Updated